runtime/connectors/gcs/gcs.go

```diff
@@ -103,3 +114,35 @@ func (c connector) ConsumeAsIterator(ctx context.Context, env *connectors.Env, s
 	}
 	return rillblob.NewIterator(ctx, bucketObj, opts)
 }
+
+func resolvedCredentials(ctx context.Context, env *connectors.Env) (*google.Credentials, error) {
+	useHostCred := env.Variables["use_host_credentials"] != "false" // true by default
+	if useHostCred {
+		// use host credentials
+		return gcp.DefaultCredentials(ctx)
+	}
+
+	secretJSONFile := env.Variables["gcs_credentials_file"]
+	var jsonData []byte
+	if secretJSONFile != "" {
+		// use credentials from file given by user
+		r, err := os.Open(secretJSONFile)
+		if err != nil {
+			return nil, err
+		}
+
+		bw := new(bytes.Buffer)
+		_, err = io.Copy(bw, r)
+		if err != nil {
+			return nil, err
+		}
+
+		jsonData = bw.Bytes()
+	} else {
+		// use credentials from json string provided by user
+		secretJSON := env.Variables["gcs_credentials"]
+		jsonData = []byte(secretJSON)
+	}
+
+	return google.CredentialsFromJSON(ctx, jsonData, "https://www.googleapis.com/auth/cloud-platform")
+}
```

Comment:
1. See earlier comment about "allow" vs "use" host credentials. So it should fallback to host credentials if `GCS_CREDENTIALS` is not set, instead of exiting early.
2. What about making env variable lookups case insensitive? We could add a `GetVar` func (or something) on `env`, which does a case insensitive lookup (or convert all keys to uppercase when creating the `connectors.Env` struct).
3. I don't think we should support `gcs_credentials_file` – in local, you can just set `GOOGLE_APPLICATION_CREDENTIALS`, and on cloud, it will open up a security leak